src/charm.py

```diff
@@ -13,7 +13,14 @@
 from pathlib import Path
 from typing import Dict, List, Literal, Optional, Tuple, get_args
 
-import psycopg2
+try:
+    # First platform-specific import, will fail on wrong architecture
+    import psycopg2
+except ModuleNotFoundError:
+    from fake_charm import block_on_wrong_architecture
+
+    block_on_wrong_architecture()
```

Comment:
We should rereaise the exception if we don't die in `block_on_wrong_architecture()`

Reply:
should be done now